lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -380,17 +380,30 @@ def sign_pki_certificate_signing_request(
             logger.warning("Error while signing PKI certificate: %s", e)
             return None
 
-    def create_or_update_pki_charm_role(self, role: str, allowed_domains: str, mount: str) -> None:
-        """Create a role for the PKI backend."""
+    def create_or_update_pki_charm_role(
+        self, role: str, allowed_domains: str, max_ttl: str, mount: str
+    ) -> None:
+        """Create a role for the PKI backend or update it if it already exists.
+
+        Args:
+            role: The name of the role to create or update.
+            allowed_domains: The list of allowed domains for the role.
+            max_ttl: The maximum TTL for the role.
+                Will be used as validity for the certificates issued by this role.
+                Should be a string in the format of a number with a unit such as
+                "120m", "10h" or "90d".
+            mount: The mount point of the PKI backend for which the role will be created.
+        """
         self._client.secrets.pki.create_or_update_role(
             name=role,
             mount_point=mount,
             extra_params={
                 "allowed_domains": allowed_domains,
                 "allow_subdomains": True,
+                "max_ttl": max_ttl,
```

Comment:
Here we set the max_ttl, what's the actual ttl that's being used when signing certificates, is it the same value?

Reply:
We sign certificates using `client.secrets.pki.sign_certificate` that could take a `ttl` in its `extra_params` and that would be the validity of the signed certificate. However this can't exceed the `max_ttl` that is set for the role of the issuer. Additionally if no `ttl` is specified in `client.secrets.pki.sign_certificate` then the `max_ttl` of the role will be used by default.

I decided to manage this centrally when the PKI engine is being configured but I improved the documentation to make sure this is clear.

If you prefer to increase readability by setting this in the `client.secrets.pki.sign_certificate` too or by moving it completely there I'd be happy to do that.

Reply:
I think it would be good from a developer point of view to also set the ttl in the `sign_certificate` method, this way we know for sure what's the cert validity. Right now we have to infer and guess

Reply:
Done